wire-ios-data-model/Source/Model/Conversation/ZMConversation+Federation.swift

```diff
@@ -29,4 +29,58 @@ extension ZMConversation {
         return domain != userDomain
     }
 
+    static func groupConversations(
+        hostedOnDomain domain: String,
+        in context: NSManagedObjectContext
+    ) -> [ZMConversation] {
+        let predicate: NSPredicate = .isGroupConversation.and(.isHostedOnDomain(domain))
+        let request = sortedFetchRequest(with: predicate)
+        return context.fetchOrAssert(request: request) as? [ZMConversation] ?? []
+    }
+
+    static func groupConversations(
+        notHostedOnDomains domains: [String],
+        in context: NSManagedObjectContext
+    ) -> [ZMConversation] {
+        let predicate: NSPredicate = .isGroupConversation.and(.isNotHostedOnDomains(domains))
+        let request = sortedFetchRequest(with: predicate)
+        return context.fetchOrAssert(request: request) as? [ZMConversation] ?? []
+    }
+
+}
+
+extension NSPredicate {
```

Comment:
This should probably be a private extension.

Reply:
yes, fixed